enterprise/models.py

```diff
@@ -971,64 +971,41 @@ def enroll(self, course_run_id, mode, cohort=None, source_slug=None, discount_pe
-                "[Enroll] Trying to enroll the enterprise user [{learner_id}] in course [{course_run_id}] in "
+                "[Enroll] Trying to enroll the enterprise customer user [{ecu_id}] in course [{course_run_id}] in "
                 "[{mode}] mode".format(
-                    learner_id=self.id,
+                    ecu_id=self.id,
                     course_run_id=course_run_id,
-                    mode=mode
+                    mode=mode,
                 )
             )
 
         if not enrolled_in_course or is_upgrading:
             if cohort and not self.enterprise_customer.enable_autocohorting:
                 raise CourseEnrollmentPermissionError("Auto-cohorting is not enabled for this enterprise")
 
-            try:
-                enterprise_course_enrollment, created = EnterpriseCourseEnrollment.objects.get_or_create(
-                    enterprise_customer_user=self,
-                    course_id=course_run_id,
-                    defaults={
-                        'source': EnterpriseEnrollmentSource.get_source(source_slug)
-                    }
-                )
-            except IntegrityError:
-                # Added to try and fix ENT-2463. This can happen if the user is already a part of the enterprise
-                # because of the following:
-                # 1. (non-enterprise) CourseEnrollment data is created
-                # 2. An async task to is signaled to run after CourseEnrollment creation
-                #    (create_enterprise_enrollment_receiver)
-                # 3. Both async task and the code in the try block run `get_or_create` on
-                # `EnterpriseCourseEnrollment`
-                # 4. A race condition happens and it tries to create the same data twice
-                # Catching will allow us to continue and ensure we can still create an order for this enrollment.
-                LOGGER.exception("IntegrityError on attempt at EnterpriseCourseEnrollment for user with id [%s] "
-                                 "and course id [%s]", self.user_id, course_run_id)
             # Directly enroll into the specified track.
-            # This should happen after we create the EnterpriseCourseEnrollment
             succeeded = True
             LOGGER.info(
-                "[Enroll] Calling LMS enrollment API for user [username] in course [course_run_id] in mode "
-                "[mode]".format(
+                "[Enroll] Calling LMS enrollment API for user {username} in course {course_run_id} in "
+                " mode {mode}".format(
                     username=self.username,
                     course_run_id=course_run_id,
-                    mode=mode
+                    mode=mode,
                 )
             )
             try:
                 enrollment_api_client.enroll_user_in_course(self.username, course_run_id, mode, cohort=cohort)
             except HttpClientError as exc:
                 succeeded = False
-                if created:
```

Comment:
The `enterprise_course_enrollment` only gets created if the learner was enrolled in the course, creating/updating a `CourseEnrollment` record.